apps/server/src/adapters/OscAdapter.ts

```diff
@@ -59,17 +60,29 @@ export class OscServer implements IAdapter {
           property,
           value,
         } satisfies ChangeOptions;
+      } else if (params.length) {
+        // transform params array into a nested object with args as the value
+        // /a/b/c/d 5 will become {a:{b:{c:{d: 5}}}}
+        transformedPayload = params.reduceRight(
+          (parm, key, index) => (index === params.length - 1 ? { [key]: args } : { [key]: parm }),
+          {},
+        );
       }
 
       try {
-        // we dont reply on OSC
-        dispatchFromAdapter(
+        const reply = dispatchFromAdapter(
           path,
           {
             payload: transformedPayload,
           },
           'osc',
         );
+        //We only send simple types back to the OSC client
+        if (typeof reply.payload === 'string' || typeof reply.payload === 'number') {
+          const client = new Client(rinfo.address, rinfo.port);
```

Comment:
what is `rinfo`? could we rename it something else?

Reply:
yes `clientInfo` maybe?